src/sentry/api/serializers/models/project.py

```diff
@@ -1063,3 +1067,25 @@ def serialize(self, obj, attrs, user):
             "features": feature_list,
             "organization": {"slug": obj.organization.slug, "name": obj.organization.name},
         }
+
+
+class MinimalProjectSerializer(Serializer):
+    def get_attrs(self, item_list: Sequence[Project], user: User, **kwargs: Any):
+        environments_by_project = get_environments_by_projects(item_list)
+        memberships_by_project = get_access_by_project(item_list, user)
+        return {
+            project: {
+                "environments": environments_by_project[project.id],
+                "isMember": memberships_by_project[project]["is_member"],
+            }
+            for project in item_list
+        }
+
+    def serialize(self, obj: Project, attrs: Mapping[str, Any], user: User):
+        return {
+            "slug": obj.slug,
+            "id": obj.id,
+            "platform": obj.platform,
+            "environments": attrs["environments"],
+            "isMember": attrs["isMember"],
```

Comment:
If you're building the project option list in the UI do you also need `isBookmarked` for the current user to render the star icons?

Reply:
ya, i havn't looked too hard at the actual callsites yet to see what fields we need. I assume we'll tweak it later. 

It'll be good to get something in so we can see the perf of something small like this vs the normal project list endpoint, then we'll know if it's worth even having a 2nd endpoint or not